arch/x86/src/boundary/boundary.rs

```diff
@@ -0,0 +1,215 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+use core::fmt::Write;
+
+use x86::bits32::eflags::EFlags;
+
+use kernel::process::FunctionCall;
+use kernel::syscall::{ContextSwitchReason, Syscall, SyscallReturn, UserspaceKernelBoundary};
+use kernel::ErrorCode;
+
+use crate::interrupts::SYSCALL_VECTOR;
+use crate::segmentation::{USER_CODE, USER_DATA};
+
+use super::UserContext;
+
+/// Defines the usermode-kernelmode ABI for x86 platforms.
+pub struct Boundary(());
```

Comment:
I think this was originally intended to be a defensive coding thing, forcing clients to use `new()` instead of instantiating `Boundary` directly.

Turns out there wasn't any internal state for this struct, so I agree this is unnecessary

Reply:
Suggestion implemented in latest push